### PR TITLE
In Progress: Add more messaging experiments documentation

### PR DESCRIPTION
**Pull request by lmegviar**
## Description (optional)

- Adds additional messaging system documentation from this [Draft Overview Doc](https://docs.google.com/document/d/1wEQEXjcSZGE8VP5qLmfS9MRmbbLY02R_4Z8-TL8OG_I/edit)

### CODE REVIEW

File: docs/messaging/how-a-message-gets-shown.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- Local provider: shipped with Firefox
+- Remote settings: a remote settings bucket
+- Experiments: part of a treatment branch
+
+All messages have an associated [trigger](https://firefox-source-docs.mozilla.org/toolkit/components/messaging-system/docs/TriggerActionSchemas/) (user action targeted by the message) and targeting condition. Additionally messages have a frequency cap and they can belong to a group that also has a frequency cap.
+
+Deciding to show a message [starts with the trigger action](https://searchfox.org/mozilla-central/rev/65d4d3399afa79c8de5a0cc11752d2ba7c31edc1/browser/components/newtab/lib/ASRouter.jsm#1323), when this happens we look at all the messages matching that trigger. We filter out blocked messages or messages that reached their frequency cap (or their groups frequency cap) and we evaluate targeting returning the first message that matched targeting.
+
+This cycle can result in a message being shown (which will send an Exposure event + message specific telemetry events) or no message being shown if targeting didn’t match (but it can still result in a Reach event being sent).
+
```
Comment (punamdahiya) [inline docs/messaging/how-a-message-gets-shown.md:17]: There is some documentaion on message display logic that might be useful to cross link here https://experimenter.info/messaging/display-logic/#triggers

File: docs/messaging/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- Impression
+- Primary click
+- Secondary click (when available)
+- Dismiss
+
+Telemetry pings will include the message id, experiment and branch slug (if enrolled) and the event type.
+
+Telemetry from all messaging surfaces ([except doorhangers when not enrolled in an experiment](https://searchfox.org/mozilla-central/rev/65d4d3399afa79c8de5a0cc11752d2ba7c31edc1/browser/components/newtab/lib/TelemetryFeed.jsm#645)) includes client id in the event ping. **Special care should be taken not to associate any browsing history with messaging events: showing a message when the user visits a particular website is not allowed because it would accidentally leak information about that visit.**
+
+Additional telemetry is available when the message is delivered as an <u>experiment</u>:
```
Comment (punamdahiya) [inline docs/messaging/telemetry.md:17]: will be good to cross link telemetry documentation https://experimenter.info/telemetry/ available for experiments

File: docs/messaging/limitations.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+title: Limitations
+slug: /messaging/limitations
+---
+
+### History state of past experiments ###
+
+It is currently not possible to enroll users into experiments based on previous experiments that they might have been a part of. If multiple messaging campaigns go out as experiments targeting different surfaces it is not possible to ensure that users will not see both campaigns.
+Mitigation is possible through the use of groups for frequency capping. If the experiments happen simultaneously then using the same group can ensure we control the time it takes for a user to see both campaigns.
+
+Similarly (using targeting) it is possible to prevent a user from seeing a message if impressions from a different message are present. However once a message has been removed (or the experiment has ended) message impressions are removed. This is only possible if the two messages overlap in the system at the same time.
```
Comment (punamdahiya) [inline docs/messaging/limitations.md:12]: This is a good capture of known limitations. I believe message impressions live for experiments when part of message group , @Mardak can confirm so may be we can call out something like  'However once a message (not part of message group) has been removed (or the experiment has ended) message impressions are removed.'

**Review by punamdahiya — APPROVED** [review]

File: docs/messaging/limitations.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+title: Limitations
+slug: /messaging/limitations
+---
+
+### History state of past experiments ###
+
+It is currently not possible to enroll users into experiments based on previous experiments that they might have been a part of. If multiple messaging campaigns go out as experiments targeting different surfaces it is not possible to ensure that users will not see both campaigns.
+Mitigation is possible through the use of groups for frequency capping. If the experiments happen simultaneously, then using the same group can ensure we control the time it takes for a user to see both campaigns.
+
+Similarly (using targeting) it is possible to prevent a user from seeing a message if impressions from a different message are present. This is only possible if the two messages overlap in the system at the same time. **Note that when using targeting or groups in this way the message *impressions* are removed once the message has been removed or the experiment has ended.**
```
Comment (lmegviar) [inline docs/messaging/limitations.md:12]: @punamdahiya how's something like this for calling out the removal of message impressions in these two cases?

**Pull request merged**